teuthology/orchestra/remote.py

```diff
@@ -199,8 +200,21 @@ def run(self, **kwargs):
 
         TODO refactor to move run.run here?
         """
-        self.ensure_online()
-        r = self._runner(client=self.ssh, name=self.shortname, **kwargs)
+        if not self.is_active():
+            if not self.reconnect():
+                raise Exception('Unable to reconnect')
+        tries = 2
```

Comment:
this is funny. why do you think we should try twice?

Reply:
@tchaikov on my "testing stand" if I have an open connection and reboot a remote host in 100% cases next try to call self.ssh.get_transport().is_active() returns True, and correspondingly a try to call paramiko execute method on existing session throws an EOFError exception. So this two tries to correctly handle this situation:
1) try run
2) if it fails, and node is online, this means it was not able to actually an issue the command on the remote host, then we need just reestablish a connection and try again
3) if the second try is failed, then their are some another issue, and let the user to handle it as required, like in reboot method, wait and try again.

I am actually not sure that this would work well, maybe I need to think about finding a use case when the first call of _run could possible produce EOFError when the command actually started to execute and then failed